### PR TITLE
Allow exceptions

### PR DESCRIPTION
**Pull request by rowleya**
This allows exceptions in the list, as demonstrated by the "GitHub Actions" exception currently in there.

Helps to fix #61.

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -3,7 +3,7 @@ name: Run tests
 
 on:
   workflow_dispatch:
-  pull_request:
+  push:
```
Comment (ianhinder) [inline .github/workflows/test.yml:6]: Thanks for the PR!  Can this be changed back to pull_request?  "pull_request" causes the tests to be run on the result of merging the branch into master, whereas "push" only tests the branch.  The tests should be rerun on every push to the PR branch, and for draft PRs, so I think it's better to have "pull_request".  Open to opinions, though!

Also, for testing things like this, you can also push an empty commit: "git commit --allow-empty -m 'Test github workflow' && git push".

**Review by ianhinder — CHANGES_REQUESTED** [review]

File: .github/workflows/test.yml (hunk 1 above)
Comment (ianhinder) [reply .github/workflows/test.yml:6]: Yes, push means every push is tested, which could be an advantage.

Before I wrote the comment, I also wasn't 100% sure that github pull-request tests were run on the result of the merge, so I went and checked.  The documentation is a little vague.  It says

> Note that GITHUB_SHA for this event is the last merge commit of the pull request merge branch. 
-- https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request

I can't find a clear definition of "pull request merge branch", but it sounds like the result of merging the PR branch into the base branch.  

Empirically, I can see that the tests are run against a special branch:

image
-- https://github.com/UoMResearchIT/RSESkillsGraph/actions/runs/6554977272/job/17802772297#step:2:58

(that was for a different branch, that still has pull_request).

See https://github.com/actions/checkout/issues/15#issuecomment-524093065 where this is confirmed.

Comment (rowleya) [reply .github/workflows/test.yml:6]: Ah OK well spotted on that one - I had assumed it didn't have that feature!  Can we have both?  There is only one way to find out!

Comment (rowleya) [reply .github/workflows/test.yml:7]: OK, so with both we get both tests it seems, which I suppose is a bit overkill but then also costs us nothing (especially on a public project)

Comment (ianhinder) [reply .github/workflows/test.yml:7]: Sure, let's have both.

**Review by ianhinder — APPROVED** [review]

File: .github/workflows/test.yml (hunk 1 above)
Comment (ianhinder) [reply .github/workflows/test.yml:7]: Hmm.  Not only do both tests runs, but they both appear on the PR conversation tab.  It's a bit annoying.  Let's go with this and see how we like it in practice.
